### ISSUE TITLE
.sav Sound Effect Index can cause GB studio to crash

### ISSUE THREAD

**Issue by 99bunnies** [chrismaltby/gb-studio#1570]
**Describe the bug**
I asume this also happens with other .sav files but i only have one to test. The Tronimal_Sound_Effects.sav included with the sample project seems to only have 9 channels for the index. When going above that number, GB studio will encounter the following error: 

_" Error invoking remote method 'sfx:play-fxhammer': Error: No channels for FX Hammer D:\game development\GB Studio Projects\SampleProject\assets\sounds\Tronimal_Sound_Effects.sav Effect 10 "_

**To Reproduce**

1. Go to the Sound Effects tab / Ctrl+5.
2. Click on "Tronimal_Sound_Effects.sav", included in the new sample project template.
3. In the 'Effect Index' box, going past '9' will cause cause GB studio to encounter an error. (in the screenshot, i've set the Effect Index number to '10' )
![gbstudio error_reproduction](https://github.com/user-attachments/assets/96986b05-4603-4bc3-8ee4-6517b88d9e3f)

4. Pressing play will produce the following error:
![gbstudio error](https://github.com/user-attachments/assets/a25d5507-c099-4c6d-b8e7-ee51c090993e)

**Expected behavior**
one of these:
- dont play a sound when you've exceeded the amount of channels in the effect index.
- a popup that says something like "empty channel" or "no effect"
- clamp the effect index so it won't go higher than the number of channels in the .sav 

**Platform (please complete the following information):**
 - OS: Windows 10 x64bit
 - App version 4.1 + 4.1.1 

**Additional context**
I was still able to save my project after I received this error screen, but the Windows tab still showed '(Modified)' next to the project name, causing it to appear like it wasn't saved.

**Comment by chrismaltby** [comment]
I actually just came across this one myself yesterday! I've just pushed a fix which will go into the next release.

I've combined both your first and last suggestions for expected behaviour. On project load the number of effects is read from the FX Hammer files and that input is now clamped to the available range. If you try to add a script that goes outside of the allowed range it will clamp to play the last effect in the file instead. Then I've also added code where the error was occurring so if there's anywhere I've missed fixing it will just not play any sound rather than crashing.
